apps/web-api/src/member-experiences/member-experiences.controller.ts

```
@@ -0,0 +1,90 @@
+import { apiMemberExperiences } from '../../../../libs/contracts/src/lib/contract-member-experience';
+
+const server = initNestServer(apiMemberExperiences);
+type RouteShape = typeof server.routeShapes;
+
+@Controller()
+export class MemberExperiencesController {
+  constructor(private readonly memberExperiencesService: MemberExperiencesService,
+  ) {}
+
+  @Api(server.route.createMemberExperience)
+  @UsePipes(ZodValidationPipe)
+  @UseGuards(UserTokenValidation)
+  async create(
+    @Body() body: CreateMemberExperienceDto
+  ){
+    const experience = await this.memberExperiencesService.create(body as any);
+    return experience;
+  }
+
+  @Api(server.route.getMemberExperience)
+  @ApiParam({ name: 'uid', type: 'string' })
+  @ApiOkResponseFromZod(ResponseMemberExperienceWithRelationsSchema)
+  async findOne(
+    @ApiDecorator() { params: { uid } }: RouteShape['getMemberExperience']
+  ) {
+    const experience = await this.memberExperiencesService.findOne(uid);
+    if (!experience) {
+      throw new NotFoundException(`Member experience not found with uid: ${uid}.`);
+    }
+    return experience;
+  }
+
+  @Api(server.route.getMemberExperienceByMemberUid)
+  @ApiParam({ name: 'memberUid', type: 'string' })
+  @ApiOkResponseFromZod(ResponseMemberExperienceWithRelationsSchema)
+  async getMemberExperienceByMemberUid(@ApiDecorator() { params: { memberUid } }: RouteShape['getMemberExperienceByMemberUid']) {
+    return await this.memberExperiencesService.getAllMemberExperience(memberUid);
+  }
+
+  @Api(server.route.updateMemberExperience)
+  @ApiParam({ name: 'uid', type: 'string' })
+  @ApiOkResponseFromZod(ResponseMemberExperienceSchema)
+  @UseGuards(UserTokenValidation)
+  @UsePipes(ZodValidationPipe)
+  async update(
+    @ApiDecorator() { params: { uid }, body }: RouteShape['updateMemberExperience'],
+    @Req() req
+  ) {
+    const experience = await this.memberExperiencesService.findOne(uid);
```

No need to have another findOne, update itself should throw the error

---

i have addressed this issue and updated the code